pyqtorch/circuit.py

```diff
@@ -74,15 +75,29 @@ def expectation(
             )
 
     @property
-    def _device(self) -> torch.device:
-        try:
-            (_, buffer) = next(self.named_buffers())
-            return buffer.device
-        except StopIteration:
-            return torch.device("cpu")
-
-    def init_state(self, batch_size: int = 1) -> torch.Tensor:
+    def _device(self) -> device:
+        device_lst: list[device] = []
+        for op in self.operations:
+            if isinstance(op, QuantumCircuit):
+                device_lst.append(op._device)
+            elif isinstance(op, Module):
+                device_lst += [b.device for b in op.buffers()]
+        maybe_device = list(set(device_lst))
```

Comment:
why not a set from the beginning?
I think the rest of the code will also work with sets too?

Although I'm a bit concerned if `b.device` could be a str? or would it by then be guaranteed to be a torch.device? set doesnt necessarily behave as expected with strings

Reply:
yes, that I understand.
But why use first a `list`, then a `set`, and then a `list` again?
I think you'll be fine with a single `set` the whole time.

Reply:
good point

Reply:
its a set now

Reply:
```
In [1]: myset = set()

In [2]: myset.add('bla')

In [3]: myset.add('bla')

In [4]: myset
Out[4]: {'bla'}
```

Reply:
neat